### PR TITLE
Lang dialog new

### PR DESCRIPTION
**Pull request by MMRoopa**
Added Language Dialog

### CODE REVIEW

File: lib/langDialog/lang_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                  fontFamily: 'Poppins',
+                ),
+                textAlign: TextAlign.center,
+                maxLines: 1,
+              ),
+            ),
+            Expanded(
+              child: Container(
+                margin: const EdgeInsets.only(
+                    left: 40.0, top: 20.0, right: 40.0, bottom: 20.0),
```
Comment (Shreyas-Mith) [inline lib/langDialog/lang_dialog.dart]: LTRB = (15.0, 20, 15.0, 20.0),

File: lib/langDialog/lang_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                child: GridView.builder(
+                    physics: const ScrollPhysics(),
+                    controller: controller,
+                    gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                      crossAxisCount: 2,
+                      mainAxisSpacing: 10,
+                      crossAxisSpacing: 15,
+                      childAspectRatio: (MediaQuery.of(context).size.width /
+                          (MediaQuery.of(context).size.height / 4.5)),
+                    ),
```
Comment (Shreyas-Mith) [inline lib/langDialog/lang_dialog.dart:88]: how is this ratio calculated?

File: lib/langDialog/lang_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+    _scrollToIndex(lastSelectedIndex);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
```
Comment (Shreyas-Mith) [inline lib/langDialog/lang_dialog.dart:47]: Please set min font where you use autosize text. min font should be 8 less that the actual size and also set max lines

File: lib/langDialog/lang_dialog.dart (hunk 2 above)
Comment (Shreyas-Mith) [reply lib/langDialog/lang_dialog.dart:88]: try 1/0.6

File: lib/langDialog/lang_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  const LangDialog({Key? key}) : super(key: key);
+
+  @override
+  _LangDialogState createState() => _LangDialogState();
+}
+
+class _LangDialogState extends State<LangDialog> {
+  final AutoScrollController controller = AutoScrollController();
+
+  final List<String> _list = [
```
Comment (Shreyas-Mith) [inline lib/langDialog/lang_dialog.dart:17]: use this 
[
    'Hindi',
    'Bengali',
    'Punjabi',
    'Kannada',
    'Tamil',
    'Telugu',
    'Marathi',
    'Urdu',
    'Malayalam',
    'Others'
  ];

File: lib/langDialog/lang_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+                                margin: const EdgeInsets.all(3.0),
+                                color: checked
+                                    ? const Color(0xFF2DAD9D)
+                                    : const Color(0xFFFFF8C9),
+                                shape: const RoundedRectangleBorder(
+                                  borderRadius:
+                                      BorderRadius.all(Radius.circular(15.0)),
+                                ),
+                                child: Center(
+                                  child: AutoSizeText(
```
Comment (Shreyas-Mith) [inline lib/langDialog/lang_dialog.dart]: add a padding of 8 (left and right)

**Review by Shreyas-Mith — CHANGES_REQUESTED** [review]

File: lib/langDialog/lang_dialog.dart (hunk 1 above)
Comment (MMRoopa) [reply lib/langDialog/lang_dialog.dart]: Done

File: lib/langDialog/lang_dialog.dart (hunk 2 above)
Comment (MMRoopa) [reply lib/langDialog/lang_dialog.dart:88]: set to 1/0.6

File: lib/langDialog/lang_dialog.dart (hunk 3 above)
Comment (MMRoopa) [reply lib/langDialog/lang_dialog.dart:47]: changes are done

File: lib/langDialog/lang_dialog.dart (hunk 4 above)
Comment (MMRoopa) [reply lib/langDialog/lang_dialog.dart:17]: updated the list

File: lib/langDialog/lang_dialog.dart (hunk 5 above)
Comment (MMRoopa) [reply lib/langDialog/lang_dialog.dart]: padding added

**Pull request merged**